### PR TITLE
Replacing Jackson Factory with Gson Factory

### PR DESCRIPTION
**Pull request by cutiepie-10**
This pull request fixes [#32130].
Tried to remove the bugs in the previous pull request.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#Sat Aug 10 17:06:24 IST 2024
```
Comment (damccorm) [inline gradle/wrapper/gradle-wrapper.properties:1]: Looks like this was added by mistake, could you please remove it?

File: sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/Transport.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
 import com.google.api.client.http.HttpRequestInitializer;
 import com.google.api.client.http.HttpTransport;
 import com.google.api.client.json.JsonFactory;
-import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.json.gson.GsonFactory;
```
Comment (damccorm) [inline sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/Transport.java:26]: From the precommits (https://github.com/apache/beam/actions/runs/10342154630/job/28624760220?pr=32148), I see the following error:

```
Execution failed for task ':sdks:java:extensions:google-cloud-platform-core:analyzeClassesDependencies'.
> Dependency analysis found issues.
  usedUndeclaredArtifacts
   - com.google.http-client:google-http-client-gson:1.44.1@jar
  unusedDeclaredArtifacts
   - com.google.http-client:google-http-client-jackson2:1.44.1@jar
```

I think we need to replace the jackson artifact with the gson one here - https://github.com/apache/beam/blob/dd3fa04c90df95dd9998264f007f7b84bb18fe34/sdks/java/extensions/google-cloud-platform-core/build.gradle#L42

(example where we've already included the gson library - https://github.com/apache/beam/blob/780eef98083fe56f81cc5c62dc8ff193993584f0/sdks/java/io/solace/build.gradle#L50 )

**Review by damccorm — COMMENTED** [review]
Thanks! This looks pretty good, I had a couple of comments though

**Comment by liferoad** [comment]
We should update https://github.com/apache/beam/blob/master/CHANGES.md.

cc @Abacn

**Comment by damccorm** [comment]
Hey @cutiepie-10 we can keep this PR open, it just needs a few minor changes!

**Comment by cutiepie-10** [comment]
Hi @damccorm ,
Please checkout the new pull request on #32130. It has the latest changes. Actually by mistake, I made commit on the other branch and this branch could not detect new changes. It is my request to check the other branch.

**Comment by damccorm** [comment]
No problem

**Pull request closed without merging**